101-servicebus-topic-subscription/azuredeploy.json

```diff
@@ -1,62 +1,102 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
   "contentVersion": "1.0.0.0",
   "parameters": {
     "serviceBusNamespaceName": {
       "type": "string",
       "metadata": {
-        "description": "Name of the Service Bus Namespace"
+        "description": "Name of the Service Bus namespace"
       }
     },
-    "serviceBusTopicName": {
+    "serviceBusTopicName1": {
       "type": "string",
       "metadata": {
-        "description": "Name of the Service Bus Topic"
+        "description": "Name of the Topic"
       }
     },
-    "serviceBusTopicSubscriptionName": {
+    "serviceBusTopicName2": {
       "type": "string",
       "metadata": {
-        "description": "Name of the Service Bus Topic Subscription"
+        "description": "Name of the Topic"
+      }
+    },
+    "serviceBusSubscriptionName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the Subscription"
+      }
+    },
+    "serviceBusApiVersion": {
+      "type": "string",
+      "defaultValue": "2017-04-01",
+      "metadata": {
+        "description": "Service Bus ApiVersion used by the template"
       }
     }
   },
   "variables": {
-    "sbVersion": "2015-08-01"
+    "defaultSASKeyName": "RootManageSharedAccessKey",
+    "authRuleResourceId": "[resourceId('Microsoft.ServiceBus/namespaces/authorizationRules', parameters('serviceBusNamespaceName'), variables('defaultSASKeyName'))]"
   },
   "resources": [
     {
-      "apiVersion": "2015-08-01",
+      "apiVersion": "[parameters('serviceBusApiVersion')]",
```

Comment:
Use a literal apiVersion so schema validation can be done.

Reply:
Resolved